### PR TITLE
Return null for aggregate functions if quantity units are incompatible

### PR DESCRIPTION
**Pull request by losborne**
From the email thread concerning `Sum`, this PR returns `null` for aggregate functions when Quantity units are not compatible, e.g.:
```
Sum({9000 '/d', 200 'mg/d'}) = null
```

For `Min` and `Max`, the units are checked to be compatible, but the original unit is returned rather than the converted unit.

For `Interval` `start of` and `end of` operators it now ensures the values are in the context's timezoneOffset if they are maximum or minimum DateTime.

Pull requests into cql-execution require the following.
Submitter and reviewer should ✔ when done.
For items that are not-applicable, mark "N/A" and ✔.

[CDS Connect](https://cds.ahrq.gov/cdsconnect) and [Bonnie](https://github.com/projecttacoma/bonnie) are the main users of this repository. 
It is strongly recommended to include a person from each of those projects as a reviewer.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Code diff has been done and been reviewed (it does not contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] Code coverage has not gone down and all code touched or added is covered.
- [x] All dependent libraries are appropriately updated or have a corresponding PR related to this change
- [x] `cql4browsers.js` built with `yarn run build-everything` if coffeescript source changed.

**Reviewer:**

Name: @cmoesel 
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

Name: @hossenlopp 
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: src/elm/aggregate.coffee [hunk 1]
```diff
@@ -45,6 +49,14 @@ module.exports.Min = class Min extends AggregateExpression
     list = @source.execute(ctx)
     return null unless list?
     listWithoutNulls = removeNulls(list)
+
+    # Check for incompatible units and return null. We don't want to convert
+    # the units for Min/Max, so we throw away tue converted array if it succeeds
```
Comment (cmoesel) [inline src/elm/aggregate.coffee]: Small typo: `tue` -> `the`

File: src/elm/aggregate.coffee [hunk 2]
```diff
@@ -60,6 +72,14 @@ module.exports.Max = class Max extends AggregateExpression
     items = @source.execute(ctx)
     return null unless items?
     listWithoutNulls = removeNulls(items)
+
+    # Check for incompatible units and return null. We don't want to convert
+    # the units for Min/Max, so we throw away tue converted array if it succeeds
```
Comment (cmoesel) [inline src/elm/aggregate.coffee]: Same typo: `tue` -> `the`

File: src/elm/aggregate.coffee (hunk 1 above)
Comment (losborne) [reply src/elm/aggregate.coffee]: Good catch! Fixed this one and the one below just now

File: src/elm/interval.coffee [hunk 3]
```diff
@@ -2,7 +2,7 @@
 { ThreeValuedLogic } = require '../datatypes/logic'
 { build } = require './builder'
 { Quantity, doAddition, doSubtraction, compare_units, convert_value } = require '../datatypes/quantity'
-{ successor, predecessor } = require '../util/math'
+{ successor, predecessor, maxValueForInstance, MAX_DATETIME_VALUE, minValueForInstance, MIN_DATETIME_VALUE } = require '../util/math'
```
Comment (cmoesel) [inline src/elm/interval.coffee]: Are you using `maxValueForInstance` or `minValueForInstance`?  I don't see any usages in the diff.

File: src/elm/interval.coffee [hunk 4]
```diff
@@ -137,7 +137,12 @@ module.exports.Start = class Start extends Expression
   exec: (ctx) ->
     interval = @arg.execute(ctx)
     return null if !interval?
-    return interval.start()
+    start = interval.start()
+    # fix the timezoneOffset of minimum Datetime to match context offset
+    if start?.isDateTime && start.equals(MIN_DATETIME_VALUE)
+      start.timezoneOffset = ctx.getTimezoneOffset()
```
Comment (cmoesel) [inline src/elm/interval.coffee]: So... this wouldn't survive a round-trip, would it?  In other words, now `start` does *not* equal `MIN_DATETIME_VALUE` anymore -- and if we were to reconstruct an interval with it, it wouldn't be identified as the min/max anymore.  I guess that's OK?

File: src/elm/interval.coffee (hunk 3 above)
Comment (hossenlopp) [reply src/elm/interval.coffee]: Good catch. These have been removed.

File: src/elm/interval.coffee (hunk 4 above)
Comment (cmoesel) [reply src/elm/interval.coffee]: After discussing over Slack I get it now.  No problem.

**Pull request merged**